src/modules/75-cd/components/PipelineStudio/DeployEnvSpecifications/DeployEnvSpecifications.tsx

```diff
@@ -101,7 +101,12 @@ export default function DeployEnvSpecifications(
         if (specObject) {
           // istanbul ignore else
           if (value.environment) {
-            specObject.environment = value.environment
+            specObject.environment = {
+              ...value.environment,
+              provisioner:
+                (value.environment.provisioner && stage?.stage?.spec?.environment?.provisioner) ||
+                value.environment.provisioner
```

Comment:
```suggestion
              provisioner: stage?.stage?.spec?.environment?.provisioner || value.environment?.provisioner
```
Why do we need explicit && for the first condition, can we add optional chaining for provisoners

Reply:
case (a && b) || a

a = undefined
b = 1
(a && b) || a
==> undefined (a)
'expected behavior'
-------------------
a = 1
b = 2
==> 2 (b)
(a && b) || a
'expected behavior'
-------------------
a = 1
b = undefined
(a && b) || a
===> 1
'expected behavior'


case a || b

a = 1
b = undefined

=>> 1 (a) 'expected behavior'
------------------------
a = 1
b = 2
=>> 1 (a)'expected behavior'
------------------------
a = undefined
b = 1
=>> 1 (b) 'unexpected behavior'